File: library/src/main/java/com/yy/codex/uikit/CGSize.java
package com.yy.codex.uikit;

import android.support.annotation.NonNull;

/**
 * Created by cuiminghui on 2017/1/3.
 */

public final class CGSize {

    final public double width;
    final public double height;

    public CGSize(double width, double height) {
        this.width = width;
        this.height = height;
    }

    @NonNull
    public CGSize setWidth(double width) {
        return new CGSize(width, height);
    }

    @NonNull
    public CGSize setHeight(double height) {
        return new CGSize(width, height);
    }

    @NonNull @Override
    public String toString() {
        return "CGSize{mWidth=" + width + ",height=" + height + "}";
    }

}

File: library/src/main/java/com/yy/codex/uikit/CGSize.kt
package com.yy.codex.uikit

/**
 * Created by cuiminghui on 2017/1/3.
 */

class CGSize(val width: Double, val height: Double) {

    fun setWidth(width: Double): CGSize {
        return CGSize(width, height)
    }

    fun setHeight(height: Double): CGSize {
        return CGSize(width, height)
    }

    override fun equals(other: Any?): Boolean {
        var other = other as? CGSize ?: return false
        return Math.abs(width - other.width) < 0.01 && Math.abs(height - other.height) < 0.01
    }

    override fun toString(): String {
        return "CGSize{width=$width,height=$height}"
    }

    companion object {
        val zero = CGSize(0.0, 0.0)
    }

}
